refactor(parser): Remove Lexer peeking for js/class

leaysgur opened the pull request:
Part of #11194 

Just replace, TS does the same thing this time.

> https://github.com/microsoft/TypeScript/blob/b504a1eed45e35b5f54694a1e0a09f35d0a5663c/src/compiler/parser.ts#L7730
> https://github.com/microsoft/TypeScript/blob/b504a1eed45e35b5f54694a1e0a09f35d0a5663c/src/compiler/parser.ts#L8050

Boshen commented:
### Merge activity

* **May 23, 7:50 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
refactor(parser): Remove Lexer peeking for js/class (#11243)

Part of #11194

Just replace, TS does the same thing this time.

> https://github.com/microsoft/TypeScript/blob/b504a1eed45e35b5f54694a1e0a09f35d0a5663c/src/compiler/parser.ts#L7730
> https://github.com/microsoft/TypeScript/blob/b504a1eed45e35b5f54694a1e0a09f35d0a5663c/src/compiler/parser.ts#L8050
